src/xenia/cpu/processor.cc

```diff
@@ -197,7 +197,7 @@ Function* Processor::DefineBuiltin(const std::string_view name,
   function->set_end_address(address + 4);
   function->set_name(name);
 
-  auto builtin_function = static_cast<BuiltinFunction*>(function);
+  auto builtin_function = dynamic_cast<BuiltinFunction*>(function);
```

Comment:
dynamic_cast (here and in module.cc) is only needed when you actually need to check if an object is an instance of a specific subclass — here there are no null checks afterwards, and as this function is declared in builtin_module_, it's clear that this will be a BuiltinFunction (and Symbol::Type::kFunction in module.cc also implies that it will be a Function), using static_cast is natural here.

Reply:
Rebased and removed the commit that adds dynamic cast